staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -125,7 +120,7 @@ func New(
 		lastSyncedNodes:  []*v1.Node{},
 	}
 
-	serviceInformer.Informer().AddEventHandlerWithResyncPeriod(
+	serviceInformer.Informer().AddEventHandler(
```

Comment:
I don't know, but if that is the case and my reading is correct, the event handler will not enqueue the service because it didn't change, so the external agents changes will not be reconciled

Reply:
But that's exactly what "resync period" is doing, right?
What resync period gives us (shouldn't be confused with relist) is that every <resync period> it enqueues all items that exists in its cache (and they are then picked by handlers and being processed).

Reply:
so, the way I understand it is that you want to resync the Services to go over the reconcile loop again  to see if the cloud objects that needs to "implement" the Service Loadbalancer were not modified, 

```
		UpdateFunc: func(old, cur interface{}) {
				oldSvc, ok1 := old.(*v1.Service)
				curSvc, ok2 := cur.(*v1.Service)
				if ok1 && ok2 && (s.needsUpdate(oldSvc, curSvc) || needsCleanup(curSvc)) {
					s.enqueueService(cur)
				}
```

the current event handler will not do that, because the Service is correct, maybe someone deleted a firewall rule or something, but this controller AFAIK will never detect that.

So, my change is wrong, what we need to do is to send the service again through the queue?